### PR TITLE
Initial implementation

### PR DESCRIPTION
**Pull request by jthomerson**

### CODE REVIEW

File: src/cli.js [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+
+argv.slaves = _.map(argv.slaves, mapTableName.bind(null, 'slave'));
+
+if (_.isEmpty(argv['starting-key'])) {
+   argv['starting-key'] = undefined;
+} else {
+   if (argv.parallel) {
+      console.log('ERROR: --starting-key can not be used when using --parallel');
+      console.log('because each segment would need its own starting key.');
+      argsFailed = 1;
```
Comment (onebytegone) [inline src/cli.js]: The other instances where `argsFailed` is adjusted, the code is using `true`. Is there a particular reason this needs to be `1`?

File: src/cli.js [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+
+argv = minimist(process.argv.slice(2), argvOpts);
+
+function mapTableName(type, name) {
+   var parts = name.split(':');
+
+   if (parts.length !== 2) {
+      console.log('Your table name must be supplied in two parts: "<region>:<table-name>"');
+      console.log('This %s table does not meet that requirement:', type, name);
+      argsFailed = true;
```
Comment (onebytegone) [inline src/cli.js]: I understand why this is here in the function, but I'm a tad wary of doing so. It gives `mapTableName` two jobs, validate/parse the table name and update the script's validation state. In the long run, this is probably fine. However, just want to be sure.

File: src/cli.js [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+      if (_.isEmpty(argv['mfa-serial'])) {
+         console.log('Assuming role %s', params.RoleArn);
+      } else {
+         params.SerialNumber = argv['mfa-serial'];
+         params.TokenCode = argv['mfa-token'];
+         console.log('Assuming role %s with MFA %s (%s)', params.RoleArn, params.SerialNumber, params.TokenCode);
+      }
+
+      AWS.config.credentials = new AWS.TemporaryCredentials(params);
+      // See jthomerson comments on https://github.com/aws/aws-sdk-js/issues/1064
```
Comment (onebytegone) [inline src/cli.js]: Should we also note this ticket too? https://github.com/aws/aws-sdk-js/issues/1664

File: .eslintrc.json [hunk 4]
```diff
@@ -1,5 +1,9 @@
 {
 
-   "extends": "eslint-config-silvermine/node"
+   "extends": "eslint-config-silvermine/node",
+
+   "rules": {
+      "no-console": 0
```
Comment (onebytegone) [inline .eslintrc.json]: `"off"`? e.g. https://github.com/silvermine/dynamodb-capacity-manager/blob/e18faa52e2dbf091087f31796f3bb10dd2f87f3b/.eslintrc.json#L6

File: src/cli.js (hunk 2 above)
Comment (jthomerson) [inline src/cli.js]: Basically, this whole script is just procedural code for arg validation, so I don't feel it really needs to be engineered to the degree you would for some nice, clean object. If it's easy to read and works well, it did its job - at least that's my philosophy.

File: src/cli.js (hunk 1 above)
Comment (jthomerson) [inline src/cli.js]: I have no idea why or how that got there.

File: src/cli.js (hunk 3 above)
Comment (jthomerson) [inline src/cli.js]: Good catch. I meant to go back and add that after creating the ticket and forgot.

File: .eslintrc.json (hunk 4 above)
Comment (jthomerson) [inline .eslintrc.json]: Funny, I had initially put `false`, and then got this error:

```
Configuration for rule "no-console" is invalid:
Severity should be one of the following: 0 = off, 1 = warn, 2 = error (you passed 'false').
```

So then I put zero and forgot to look what we used elsewhere. Apparently they're equivalent. Changed anyway to be consistent.

**Comment by jthomerson** [comment]
@onebytegone back to you with the requested changes.

File: src/cli.js (hunk 2 above)
Comment (onebytegone) [inline src/cli.js]: 👍 That sounds reasonable

**Pull request merged**